test_rules_db_from_device is flaky

One some of our machines the `test_rules_db_from_device` test is flaky, possibly racy, executed in a diffferent order than required. It tries to access a table that does not yet exist.

Happens on pywemo 0.6.4.

```
__________________________ test_rules_db_from_device ___________________________

temp_file = <tempfile._TemporaryFileWrapper object at 0x7ffff4fc6af0>
sqldb = <sqlite3.Connection object at 0x7ffff4473120>

    def test_rules_db_from_device(temp_file, sqldb):
        rules_db.RulesRow(RuleID=501, Name="", Type="").update_db(sqldb.cursor())
        sqldb.commit()
        sqldb.close()
        zip_content = base64.b64decode(rules_db._pack_db(temp_file, "inner.db"))
        mock_response = create_autospec(urllib3.HTTPResponse, instance=True)
        mock_response.status = 200
        mock_response.data = zip_content
        store_rules = []
    
        class Device:
            name = MOCK_NAME
            udn = MOCK_UDN
            session = Session("http://localhost/")
    
            class rules:
                @staticmethod
                def FetchRules():
                    return {
                        "ruleDbVersion": "1",
                        "ruleDbPath": "http://localhost/rules.db",
                    }
    
                @staticmethod
                def StoreRules(**kwargs):
                    store_rules.append(kwargs)
    
        with patch(
            "urllib3.PoolManager.request", return_value=mock_response
        ) as mock_request:
>           with rules_db.rules_db_from_device(Device) as db:

tests/ouimeaux_device/api/unit/test_rules.py:246: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/d44wd6n98f93hjr6q1d1phhh1hw7a17d-python3-3.8.8/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
pywemo/ouimeaux_device/api/rules_db.py:385: in rules_db_from_device
    rules = RulesDb(
pywemo/ouimeaux_device/api/rules_db.py:175: in __init__
    self._rules = _index_by_primary_key(RulesRow.select_all(cursor))
pywemo/ouimeaux_device/api/rules_db.py:430: in _index_by_primary_key
    for row in rows:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pywemo.ouimeaux_device.api.rules_db.RulesRow'>
cursor = <sqlite3.Cursor object at 0x7ffff59ac6c0>

    @classmethod
    def select_all(cls, cursor):
        """Select all Row entries from the underlying sqlite table."""
        names = ",".join(cls.FIELDS.keys())
>       cursor.execute(f"SELECT {names} FROM {cls.TABLE_NAME}")
E       sqlite3.OperationalError: no such table: RULES

pywemo/ouimeaux_device/api/db_orm.py:67: OperationalError
```

--- a/pywemo/ouimeaux_device/api/rules_db.py
+++ b/pywemo/ouimeaux_device/api/rules_db.py
@@ -410,6 +410,7 @@ def _unpack_db(content, db_file):
         inner_file_name = zip_file.namelist()[0]
         with zip_file.open(inner_file_name) as zipped_db_file:
             db_file.write(zipped_db_file.read())
+            db_file.flush()
         return inner_file_name
     raise RuntimeError("Could not find database within zip file")
 

